### ISSUE TITLE
files.watcherExclude setting should extend files.exclude

### ISSUE THREAD

**Issue by kvdveer** [microsoft/vscode#76577]
*TL;DR*: using inotify watches on excluded files is probably not useful, and rather wasteful.

I'm working on a small project that needs to run on a wide variety of different python interpreters. To ensure compatibility with the different interpreters, I use `tox`, a tool which build different virtualenvs for each interpeter. These virtualenvs are stored in `.tox/`, relative to the project root (or wherever tox.ini is placed).

Of course, the `.tox/` directory is of little use to the developer, and e.g. search results from this directory are not useful. To reduce clutter, I added `.tox` to the `files.exclude` setting. This works fine.

After a system reinstall I had not yet applied the `max_user_watches` sysctl. To my surprise, vscode complained that it could not create enough file watches; despite the fact that only a dozen files were visible in the file explorer. After some investigation, it turned out that vscode attempted to set up file watches for all files in the `.tox/` directory. That's several hundreds of thousands of files. Of course, that's rather easilly mitigated by using the `files.watcherExclude` setting, but I'd argue that watching files that were explicitly excluded from the project is a waste of resources.
